cmd/run.go

```diff
@@ -622,246 +622,264 @@ var runCmd = &cobra.Command{
-						// after the StopGracefully function is called.
-						<-stopChan
-						os.Exit(gerr.FailedToCreateClient)
-					}()
-					StopGracefully(
-						runCtx,
-						nil,
-						metricsMerger,
-						metricsServer,
-						pluginRegistry,
-						logger,
-						servers,
-						stopChan,
-						httpServer,
-						grpcServer,
-					)
 				}
-			}
 
-			// Verify that the pool is properly populated.
-			logger.Info().Fields(map[string]interface{}{
-				"name":  name,
-				"count": strconv.Itoa(pools[name].Size()),
-			}).Msg("There are clients available in the pool")
-
-			if pools[name].Size() != currentPoolSize {
-				logger.Error().Msg(
-					"The pool size is incorrect, either because " +
-						"the clients cannot connect due to no network connectivity " +
-						"or the server is not running. exiting...")
-				pluginRegistry.Shutdown()
-				os.Exit(gerr.FailedToInitializePool)
-			}
+				// Verify that the pool is properly populated.
+				logger.Info().Fields(map[string]interface{}{
+					"name":  configBlockName,
+					"count": strconv.Itoa(pools[configGroupName][configBlockName].Size()),
+				}).Msg("There are clients available in the pool")
+
+				if pools[configGroupName][configBlockName].Size() != currentPoolSize {
+					logger.Error().Msg(
+						"The pool size is incorrect, either because " +
+							"the clients cannot connect due to no network connectivity " +
+							"or the server is not running. exiting...")
+					pluginRegistry.Shutdown()
+					os.Exit(gerr.FailedToInitializePool)
+				}
 
-			pluginTimeoutCtx, cancel = context.WithTimeout(
-				context.Background(), conf.Plugin.Timeout)
-			defer cancel()
+				pluginTimeoutCtx, cancel = context.WithTimeout(
+					context.Background(), conf.Plugin.Timeout)
+				defer cancel()
 
-			_, err = pluginRegistry.Run(
-				pluginTimeoutCtx,
-				map[string]interface{}{"name": name, "size": currentPoolSize},
-				v1.HookName_HOOK_NAME_ON_NEW_POOL)
-			if err != nil {
-				logger.Error().Err(err).Msg("Failed to run OnNewPool hooks")
-				span.RecordError(err)
+				_, err = pluginRegistry.Run(
+					pluginTimeoutCtx,
+					map[string]interface{}{"name": configBlockName, "size": currentPoolSize},
+					v1.HookName_HOOK_NAME_ON_NEW_POOL)
+				if err != nil {
+					logger.Error().Err(err).Msg("Failed to run OnNewPool hooks")
+					span.RecordError(err)
+				}
 			}
 		}
 
 		span.End()
 
 		_, span = otel.Tracer(config.TracerName).Start(runCtx, "Create proxies")
 		// Create and initialize prefork proxies with each pool of clients.
-		for name, cfg := range conf.Global.Proxies {
-			logger := loggers[name]
-			clientConfig := clients[name]
-			// Fill the missing and zero value with the default one.
-			cfg.HealthCheckPeriod = config.If(
-				cfg.HealthCheckPeriod > 0,
-				cfg.HealthCheckPeriod,
-				config.DefaultHealthCheckPeriod,
-			)
+		for configGroupName, configGroup := range conf.Global.Proxies {
+			for configBlcokName, cfg := range configGroup {
```

Comment:
Consider renaming all instances of this type:
```suggestion
			for configBlockName, cfg := range configGroup {
```